launcher/src/main/java/io/snappydata/tools/QuickLauncher.java

```diff
@@ -125,15 +125,36 @@ private int start(final String[] args) throws IOException, InterruptedException
     // add java path to command-line first
     commandLine.add(System.getProperty("java.home") + "/bin/java");
     commandLine.add("-server");
-    // https://github.com/airlift/jvmkill is added to libgemfirexd.so
-    // adding agentpath helps kill jvm in case of OOM. kill -9 is not
-    // used as it fails in certain cases
-    if (Platform.isLinux()) {
-      if (Platform.is64Bit()) {
-        commandLine.add("-agentpath:" + snappyHome + "/jars/libgemfirexd64.so");
-      } else {
-        commandLine.add("-agentpath:" + snappyHome + "/jars/libgemfirexd.so");
+
+    try {
+      // https://github.com/airlift/jvmkill is added to libgemfirexd.so
+      // adding agentpath helps kill jvm in case of OOM. kill -9 is not
+      // used as it fails in certain cases
+      if (Platform.isLinux()) {
+        if (Platform.is64Bit()) {
+          String agentPath = snappyHome + "/jars/libgemfirexd64.so";
+          System.load(agentPath);
+          commandLine.add("-agentpath:" + agentPath);
+        } else {
+          String agentPath = snappyHome + "/jars/libgemfirexd.so";
+          System.load(agentPath);
+          commandLine.add("-agentpath:" + agentPath);
+        }
+      } else if (Platform.isMac()) {
+        if (Platform.is64Bit()) {
+          String agentPath = snappyHome + "/jars/libgemfirexd64.dylib";
+          System.load(agentPath);
+          commandLine.add("-agentpath:" + agentPath);
+        } else {
+          String agentPath = snappyHome + "/jars/libgemfirexd.dylib";
+          System.load(agentPath);
+          commandLine.add("-agentpath:" + agentPath);
+        }
       }
+    } catch (UnsatisfiedLinkError e) {
+      System.out.println("WARNING: agent not loaded. service might not be killed on OutOfMemory");
+    }catch (SecurityException e){
+      System.out.println("WARNING: agent not loaded. service might not be killed on OutOfMemory");
```

Comment:
Java8 allows combining common code for exceptions. Also you could print out the exception for information:

```java
} catch (UnsatisfiedLinkError | SecurityException e) {
  System.out.println("WARNING: agent not loaded due to " + e +
     ". Service might not be killed on OutOfMemory. Build jvmkill.c on your platform using build.sh script" +
     " from source on your platform and replacing the library in product jars directory to enable the agent.");
}
```